src/flamegraph/color.rs

```diff
@@ -0,0 +1,272 @@
+use std::str::FromStr;
+use std::collections::HashMap;
+use std::fs::OpenOptions;
+use std::io;
+use std::io::Write;
+use std::path::Path;
+use std::fs::File;
+use std::io::BufReader;
+use std::io::BufRead;
+
+pub(super) const VDGREY: &str = "rgb(160,160,160)";
+pub(super) const DGREY: &str = "rgb(200,200,200)";
+
+const YELLOW_GRADIENT: (&str, &str) = ("#eeeeee", "#eeeeb0");
+const BLUE_GRADIENT: (&str, &str) = ("#eeeeee", "#e0e0ff");
+const GRAY_GRADIENT: (&str, &str) = ("#f8f8f8", "#e8e8e8");
+
+#[derive(Debug, PartialEq)]
+pub enum Palette {
+    Hot,
+    Mem,
+    Io,
+    Wakeup,
+    Chain,
+    Java,
+    Js,
+    Perl,
+    Red,
+    Green,
+    Blue,
+    Aqua,
+    Yellow,
+    Purple,
+    Orange,
+}
+
+impl Default for Palette {
+    fn default() -> Self {
+        Palette::Hot
+    }
+}
+
+impl FromStr for Palette {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hot" => Ok(Palette::Hot),
+            "mem" => Ok(Palette::Mem),
+            "io" => Ok(Palette::Io),
+            "wakeup" => Ok(Palette::Wakeup),
+            "chain" => Ok(Palette::Chain),
+            "java" => Ok(Palette::Java),
+            "js" => Ok(Palette::Js),
+            "perl" => Ok(Palette::Perl),
+            "red" => Ok(Palette::Red),
+            "green" => Ok(Palette::Green),
+            "blue" => Ok(Palette::Blue),
+            "aqua" => Ok(Palette::Aqua),
+            "yellow" => Ok(Palette::Yellow),
+            "purple" => Ok(Palette::Purple),
+            "orange" => Ok(Palette::Orange),
+            unknown=> Err(format!("unknown color palette: {}", unknown))
+        }
+    }
+}
+
+fn namehash(name: &str) -> f32 {
+    let mut vector = 0f32;
+    let mut weight = 1f32;
+    let mut max = 1f32;
+    let mut modulo = 10;
+
+    let name = {
+        let looking_for_module_name = if name.starts_with("`") {
+            &name[1..]
+        } else {
+            name
+        };
+
+        if let Some(index) = looking_for_module_name.find("`") {
+            &looking_for_module_name[index + 1..]
+        } else {
+            name
+        }
+    };
+
+    for character in name.bytes() {
+        let i = (character % modulo) as f32;
+        vector += (i / ((modulo - 1) as f32)) * weight;
+        modulo += 1;
+        max += weight;
+        weight *= 0.70;
+
+        if modulo > 12 {
+            break
+        }
+    }
+
+    (1f32 - vector / max)
+}
+
+fn handle_java_palette(s: &str) -> Palette {
+    if s.ends_with("]") {
+        if let Some(ai) = s.rfind("_[") {
+            if s[ai..].len() == 4 {
+                let suffix_char = &s[ai+2..ai+3];
```

Comment:
This line makes me suspect that you haven't run `rustfmt` on this code. It'd be handy if you could do that so that we don't get spurious changes as parts of later commits!

Reply:
[Sorry about that](https://github.com/jonhoo/inferno/pull/32/commits/c70f667e0080b41b5fa57feab4ad34e193f8c665)